src/components/profile/ConnectedAccounts.js

```diff
@@ -0,0 +1,196 @@
+// @flow
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { Switch, Text, View } from 'react-native'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import { CeramicSDK } from '@gooddollar/ceramic-seed-sdk'
+import Web3 from 'web3'
+import { withStyles } from '../../lib/styles'
+import { Section, Wrapper } from '../common'
+
+import useTorus from '../../components/auth/torus/hooks/useTorus'
+import GoogleBtnIcon from '../../assets/Auth/btn_google.svg'
+import FacebookBtnIcon from '../../assets/Auth/btn_facebook.svg'
+import MobileBtnIcon from '../../assets/Auth/btn_mobile.svg'
+import Recaptcha from '../../components/auth/login/Recaptcha/index'
+import { useErrorDialog } from '../../lib/undux/utils/dialog'
+
+const TITLE = 'Connected Accounts'
+const sdk = new CeramicSDK('https://ceramic-clay.3boxlabs.com')
+
+const ConnectedAccounts = ({ screenProps, styles }) => {
+  const [providers, setProviders] = useState([])
+  const [authenticatorData, setAuthenticatorData] = useState({})
+
+  const initializeAccount = useCallback(async () => {
+    const privateAccountKey = JSON.parse(localStorage.getItem('GD_masterSeed'))
+    const web3 = new Web3()
+    const publicAccountKey = web3.eth.accounts.privateKeyToAccount(privateAccountKey)
+    await sdk.initialize(privateAccountKey.slice(0, privateAccountKey.length / 2), publicAccountKey.address)
+  }, [])
+  const [showErrorDialog] = useErrorDialog()
+
+  const fetchAuthProviders = useCallback(async () => {
+    await initializeAccount()
+    const metaData = await sdk.getMeta()
+    const authenticatorsObject = metaData?.state$?.state$?.value?.content?.authenticators || {}
+    setProviders(Object.values(authenticatorsObject))
+    let objectToSetAuthenticator = Object.entries(authenticatorsObject).reduce(
+      // eslint-disable-next-line no-sequences
+      (acc, [key, value]) => ((acc[value] = key), acc),
+      {},
+    )
+    setAuthenticatorData(objectToSetAuthenticator)
+  }, [initializeAccount])
+
+  const addAuthenticatedUser = useCallback(
+    async authProvider => {
+      const provider = await AsyncStorage.getItem('connectAccountsProviderLoginInitiated')
+      AsyncStorage.removeItem('connectAccountsProviderLoginInitiated')
+      if (provider) {
+        const redirectResult = await AsyncStorage.getItem('torusRedirectResult')
+        AsyncStorage.removeItem('torusRedirectResult')
```

Comment:
I can't call torus redirect here because all the info that torus.getRedirectResult requires is in the route and once we have redirected all that info is lost